dbusmock/templates/logind.py

```diff
@@ -287,3 +290,30 @@ def AddSession(self, session_id, seat, uid, username, active):
     obj_user.Set('org.freedesktop.login1.User', 'Sessions', cur_sessions)
 
     return session_path
+
+@dbus.service.method(SESSION_IFACE,
+                     in_signature='uu',
+                     out_signature='hb')
+def TakeDevice(self, major, minor):
+    sysfs_uevent_path = '/sys/dev/char/{}:{}/uevent'.format(major, minor)
+    sysfs_uevent = open(sysfs_uevent_path, 'r')
+    devname = None
+    for line in sysfs_uevent.readlines():
+        match = re.search('DEVNAME=(.*)', line)
+        if match:
+            devname = match[1]
+            break
+    sysfs_uevent.close()
+    if not devname:
+        raise dbus.exceptions.DBusException(f'Device file {major}:{minor} doesn\'t exists',
+                                            major=major, minor=minor)
+    fd = os.open('/dev/' + devname, os.O_RDWR | os.O_CLOEXEC)
+    unix_fd = dbus.types.UnixFd(fd)
+    os.close(fd)
+    return (unix_fd, False)
+
+@dbus.service.method(SESSION_IFACE,
+                     in_signature='uu',
+                     out_signature='')
+def ReleaseDevice(self, major, minor):
+    pass
```

Comment:
This is asymmetric -- if TakeDevice() creates an open fd and returns it, then ReleaseDevice should clean it up. I.e., you need to remember the fd inside the mock. 

This comment is just for completeness -- IMHO this is too specific for a generic logind mock.